In [1]:
from google.colab import auth
from google.cloud import bigquery
auth.authenticate_user()
print('Authenticated')
project_id = "gc1-rifqijulianhasyari" #GUNAKAN GCP PROJECT-ID KALIAN MASING-MASING
client = bigquery.Client(project=project_id)

Authenticated


**NAMA : RIFQI JULIAN HASYARI**
**BATCH : HCK 006**

**PROBLEM STATEMENT**


> Menganalisa apakah kota austin sudah tergolong aman berdasarkan jumlah kasus kriminal yang terjadi selama tahun 2016 dan strategi apa yang bisa dilakukan untuk mengurangi kasus kejahatan di kota austin di tahun berikutnya.



# **SOAL PENJABARAN 1**

> Tampilkan tabel crime sampai 5 baris pertama




In [2]:
Data1=client.query('''
SELECT * FROM `bigquery-public-data.austin_crime.crime` LIMIT 5

''').to_dataframe()
Data1

unique_key     address  census_tract            clearance_date  \
0   2015150483  Austin, TX           NaN 2015-01-27 12:00:00+00:00   
1  20151542027  Austin, TX           NaN 2015-07-24 12:00:00+00:00   
2  20152001758  Austin, TX           NaN 2015-08-15 12:00:00+00:00   
3  20151340282  Austin, TX           NaN 2015-06-15 12:00:00+00:00   
4  20151871268  Austin, TX           NaN 2015-10-27 12:00:00+00:00   

       clearance_status  council_district_code  \
0           Not cleared                   <NA>   
1           Not cleared                   <NA>   
2           Not cleared                   <NA>   
3  Cleared by Exception                   <NA>   
4           Not cleared                   <NA>   

                      description district  latitude  longitude location  \
0  RAPE                                  B       NaN        NaN     None   
1  RAPE                                  D       NaN        NaN     None   
2  RAPE                                  F       NaN        NaN     None   
3  RAPE                                 UK       NaN        NaN     None   
4  RAPE                                 UK       NaN        NaN     None   

  location_description primary_type                 timestamp  x_coordinate  \
0                  nan         Rape 2015-01-15 12:00:00+00:00          <NA>   
1                  nan         Rape 2015-06-03 12:00:00+00:00          <NA>   
2                  nan         Rape 2015-07-19 12:00:00+00:00          <NA>   
3                  nan         Rape 2015-05-14 12:00:00+00:00          <NA>   
4                  nan         Rape 2015-07-06 12:00:00+00:00          <NA>   

   y_coordinate  year zipcode  
0          <NA>  2015    None  
1          <NA>  2015    None  
2          <NA>  2015    None  
3          <NA>  2015    None  
4          <NA>  2015    None

Dataframe tersebut berbicara tentang kasus kejahatan yang terjadi di kota AUSTIN. Datadrame tersebut berisikan data mulai dari alamat,jenis kasus diselesaikan, jenis kasusnya bahkan sampai zipcodenya pun ada. Kode diatas menunjukan 5 data teratas yang ada pada data bigquery tersebut.

# **SOAL PENJABARAN 2**

> Apa saja jenis tindak kejahatan yang terjadi di kota Austin selama tahun 2016?




In [3]:
Data2=client.query('''
SELECT DISTINCT primary_type
FROM `bigquery-public-data.austin_crime.crime`
where EXTRACT(YEAR FROM timestamp) = 2016

''').to_dataframe()
Data2

primary_type
0         Rape
1        Theft
2      Robbery
3  Agg Assault
4     Burglary
5   Auto Theft
6       Murder

Kode diatas menunjukan jenis kejahatan yang terjadi di kota austin selama tahun 2016. Terdapat 7 jenis kejahatan yang terdiri dari rape,theft,robbery,agg assault hingga murder.

# **SOAL PENJABARAN 3**
Berapa banyak kasus tindak kejahatan berdasarkan kasus pembersihannya (clearance status) selama tahun 2016?

In [4]:
Crime = client.query('''
SELECT  clearance_status,
        COUNT(clearance_status) AS Jumlah
FROM `bigquery-public-data.austin_crime.crime`
where EXTRACT(YEAR FROM timestamp) = 2016
GROUP BY clearance_status
ORDER BY Jumlah DESC

''').to_dataframe()
Crime

clearance_status  Jumlah
0           Not cleared   29715
1     Cleared by Arrest    5013
2  Cleared by Exception    1014
3                  None       0

Kode di atas menunjukan total kejahatan yang terjadi berdasarkan jenis penyelesaiannya. Dapat dilihat pada tabel di atas bahwa kasus yang belum diselesaikan memiliki jumlah terbanyak dengan total 29715 kasus dan diikuti oleh cleared by arrest sebesar 5013 dan cleared by exxception sebesar 1014. Hal ini menunjukan diperlukannya peningkatan kualitas dari penegak hukum dalam menyelesaikan masalah karena kasus yang belum selesai memiliki angka yang sangat jauh dibandingkan yg lainnya.

# **SOAL PENJABARAN 4**

> Jenis kejahatan apa saja yang paling banyak belum terselesaikan kasusnya di tahun 2016?




In [5]:
CrimeNotFinished = client.query('''
SELECT  primary_type,
        COUNT(primary_type) AS Jumlah
FROM `bigquery-public-data.austin_crime.crime`
where EXTRACT(YEAR FROM timestamp) = 2016 AND
clearance_status = 'Not cleared'
GROUP BY primary_type
ORDER BY Jumlah DESC

''').to_dataframe()
CrimeNotFinished

primary_type  Jumlah
0        Theft   21894
1     Burglary    4440
2   Auto Theft    1540
3  Agg Assault     919
4      Robbery     560
5         Rape     361
6       Murder       1

Pada tabel diatas menunjukan jenis kasus dan total dari kasus tersebut terjadi  yang belum terselesaikan selama tahun 2016. Kasus theft/pencurian menjadi kasus yang paling banyak belum terselesaikan di tahun 2016. Dengan data ini kita dapat menentukan strategi untuk mengurangi kasus kejahatan berupa pencurian yang terjadi di tahun2 berikutnya. Salah satu strategi yang mungkin bisa dilakukan adalah dengan menambah personil kepolisian untuk melakukan patroli di sepanjang jalan khususnya saat malam hari.

# **SOAL PENJABARAN 5**

> Di bulan apa di tahun 2016 kasus pencurian (theft) sangat banyak terjadi?




In [6]:
CrimeNotFinished = client.query('''
SELECT  primary_type as Jenis_Kriminal,
        COUNT(primary_type) AS Jumlah,
        EXTRACT(month FROM timestamp) as month
FROM `bigquery-public-data.austin_crime.crime`
where EXTRACT(YEAR FROM timestamp) = 2016 AND
primary_type = 'Theft'
GROUP BY Jenis_Kriminal,month
ORDER BY Jumlah DESC

''').to_dataframe()
CrimeNotFinished

Jenis_Kriminal  Jumlah  month
0           Theft    2282      4
1           Theft    2271      3
2           Theft    2249      6
3           Theft    2235     10
4           Theft    2213      1
5           Theft    2204      8
6           Theft    2183      5
7           Theft    2163      7
8           Theft    2162     12
9           Theft    2156      9
10          Theft    2128     11
11          Theft    1995      2

# **SOAL PENJABARAN 6**


> Di distrik apa yang paling banyak terjadi kasus pencurian selama 2016?






In [7]:
District=client.query('''

SELECT district,
      Count(district) as jumlah_kasus
FROM `bigquery-public-data.austin_crime.crime`
WHERE
      EXTRACT(YEAR FROM timestamp) = 2016 AND
      primary_type = 'Theft'
GROUP BY
      district
ORDER BY
      jumlah_kasus DESC

''').to_dataframe()
District

district  jumlah_kasus
0         D          4117
1         B          3330
2         A          3096
3         F          3034
4         E          2982
5         I          2870
6         H          2734
7         G          2117
8         C          1724
9        AP           143
10       UK            81
11       88            13

Tabel di atas menunjukan distrik mana yang memiliki kasus pencurian terbanyak sepanjang tahun 2016. Data ini dapat membantu kepolisian untuk membuat strategi patroli yang tepat untuk mengurangi kasus pencurian tersebut.
Salah satu strategi yang dapat dilakukan mungkin dengan cara mengurangi personil yang berpatroli di daerah yang memiliki jumlah kasus terrendah dan memindahkan personil polisi tersebut untuk berpatroli di daerah yang memiliki kasus tertinggi.

# **SOAL PENJABARAN 7**


> Berapa lama rata-rata waktu (dalam hari) kasus tindak kejahatan 'theft' terselesaikan sejak kasus tersebut terangkat selama tahun 2016? (Hint: Hitung selisih hari dari timestamp hingga clearance date)



In [8]:
Rata_waktu =client.query('''
SELECT primary_type,
       AVG(TIMESTAMP_DIFF(clearance_date, timestamp, DAY)) AS Rata_Rata
FROM `bigquery-public-data.austin_crime.crime`
WHERE primary_type = 'Theft'
  AND clearance_status LIKE 'Cleared%'
  AND EXTRACT(YEAR FROM timestamp) = 2016
GROUP BY primary_type


''').to_dataframe()
Rata_waktu


primary_type  Rata_Rata
0        Theft  14.295328

Berdasarkan data di atas dapat dilihat kasus pencurian memiliki rata2 waktu selama 14 hari untuk terselesaikan. Hal tersebut dapat dianalisis lebih jauh lagi apakah waktu tersebut tergolong lama atau tidak. Jika memang waktu tersebut dianggap terlalu lama maka dapat mencari strategi baru untuk pembenahan agar waktu penyelesaian akan lebih cepat. Jika memang waktu tersebut dirasa sudah pantas berarti kepolisian sudah melakukan kinerja yang baik .

# **KESIMPULAN**



> Kesimpulan yang dapat diambil berdasarkan penjabaran di atas adalah kota austin memiliki jumlah kasus kriminal yang tinggi selama tahun 2016. Selain itu terdapat banyak sekali kasus yang belum terselesaikan selama tahun 2016 seperti yang dapat dilihat pada soal penjabaran 3. Hal ini dapat menjadi catatan bagi kepolisian untuk berbenah dan menyiapkan strategi guna menyambut tahun berikutnya sehingga kasus kejahatan di kota austin akan berkurang. Terdapat beberapa strategi yang mungkin bisa dilakukan seperti menambah personil kepolisian untuk melakukan patroli di district yang memiliki jumlah kasus terbanyak serta meningkatkan kinerja untuk mengurangi kasus kejahatan yang tidak terselesaikan.

